src/python/pants/engine/process_test.py

```diff
@@ -260,11 +263,21 @@ def test_interactive_process_cannot_have_input_files_and_workspace() -> None:
         InteractiveProcess(argv=["/bin/echo"], input_digest=mock_digest, run_in_workspace=True)
 
 
-def test_find_binary_non_existent(rule_runner: RuleRunner) -> None:
+def which_binary_path() -> str:
+    path = shutil.which("which")
+    if not path:
+        raise EnvironmentError("`which` not discoverable on your $PATH.")
+    return os.path.dirname(path)
+
+
+@pytest.mark.parametrize("which_path", (None, which_binary_path()))
+def test_find_binary_non_existent(rule_runner: RuleRunner, which_path: str | None) -> None:
     with temporary_dir() as tmpdir:
         search_path = [tmpdir]
+        if which_path:
+            search_path.append(which_path)
```

Comment:
This works, because `nonexistent-bin` is so unlikely to be on (say) `/usr/bin` with `which`. If you symlinked `which_path` into `tmpdir` instead of expanding the `search_path` that would be perfectly robust and you could even then `s/nonexistent-bin/python/` or something else obviously on the PATH for this test suite to even be running.